modules/auxiliary/scanner/oracle/tnspoison_checker.rb

```diff
@@ -42,8 +42,19 @@ def run_host(ip)
       send_packet = tns_packet("(CONNECT_DATA=(COMMAND=service_register_NSGR))")
       sock.put(send_packet)
       packet = sock.read(100)
+      hex_packet = Rex::Text.to_hex(packet, prefix = ':')
+      split_hex = hex_packet.split(":")
       find_packet = /\(ERROR_STACK=\(ERROR=/ === packet
-      find_packet == true ? print_error("#{ip}:#{rport} is not vulnerable ") : print_good("#{ip}:#{rport} is vulnerable")
+      #find_packet == true ? print_error("#{ip}:#{rport} is not vulnerable ") : print_good("#{ip}:#{rport} is vulnerable")
+      if find_packet == true
```

Comment:
I reviewed the logic of Oracle TNS Poisoning. As per my understanding the service should return:
 - ACCEPT (02) response for vulnerable.
 - REFUSE (04) or ERROR_STACK for not vulnerable.
 - Any other TNS packet type or non-TNS packet type response for "might not be vulnerable".

Instead of saying "not vulnerable" it says "might not be vulnerable" because I am not 100% if it is vulnerable or not and leave it to the user for further analysis.

The latest code update is below. Please feel free to let me know if it makes sense.